File: codigos/1 - estrutura sequencial/1018 - cedulas/Main.py
valor = int(input())
print(valor)

def calcular_nota(nota: int) -> None:
    global valor
    qntd: int = valor//nota
    valor -= qntd*nota
    print(f"{qntd} nota(s) de R$ {nota},00")
    

notas: list[int] = [100, 50, 20, 10, 5, 2, 1]

for nota in notas:
    calcular_nota(nota)

File: codigos/1 - estrutura sequencial/1018 - cedulas/Main.clj
(def cedulas [100 50 20 10 5 2 1])

(defn qntd-notas 
    ([n] (qntd-notas [] 0 n))
    ([qntd_cedulas i valor-atual]
        (if (= i (count cedulas))
            qntd_cedulas
            (let [qntd (quot valor-atual (cedulas i))
                valor (* qntd (cedulas i))]
                (recur (conj qntd_cedulas qntd) (inc i) (- valor-atual valor))))))

(defn main []
    (let [n (read)]
        (println n)
        (doall (map #(println (str %1 " nota(s) de R$ " %2 ",00")) (qntd-notas n) cedulas))))

(main)